File: Assets/AltUnityTester-BindingsAndExamples(can_be_deleted)/Bindings~/java/src/main/java/ro/altom/altunitytester/Commands/FindObject/AltFindObject.java
package ro.altom.altunitytester.Commands.FindObject;

import com.google.gson.Gson;
import ro.altom.altunitytester.AltBaseSettings;
import ro.altom.altunitytester.AltUnityDriver;
import ro.altom.altunitytester.AltUnityObject;
import ro.altom.altunitytester.Commands.AltBaseCommand;

public class AltFindObject extends AltBaseFindObject {
    private AltFindObjectsParameters altFindObjectsParameters;
    public AltFindObject(AltBaseSettings altBaseSettings, AltFindObjectsParameters altFindObjectsParameters) {
        super(altBaseSettings);
        this.altFindObjectsParameters = altFindObjectsParameters;
    }
    public AltUnityObject Execute(){
        if(altFindObjectsParameters.isEnabled() && altFindObjectsParameters.getBy()== AltUnityDriver.By.NAME){
            send(CreateCommand("findActiveObjectByName", altFindObjectsParameters.getValue(), altFindObjectsParameters.getCameraName(), String.valueOf(altFindObjectsParameters.isEnabled())));
        }else{
            String path= SetPath(altFindObjectsParameters.getBy(),altFindObjectsParameters.getValue());
            send(CreateCommand("findObject", path, altFindObjectsParameters.getCameraName(), String.valueOf(altFindObjectsParameters.isEnabled())));
        }
        return ReceiveAltUnityObject();
    }
}

File: Assets/AltTester/AltDriver/Commands/FindObjects/AltFindObject.cs
namespace Altom.AltDriver.Commands
{
    public class AltFindObject : AltBaseFindObjects
    {
        AltFindObjectParams cmdParams;

        public AltFindObject(IDriverCommunication commHandler, By by, string value, By cameraBy, string cameraValue, bool enabled) : base(commHandler)
        {
            cameraValue = SetPath(cameraBy, cameraValue);
            string path = SetPath(by, value);
            cmdParams = new AltFindObjectParams(path, cameraBy, cameraValue, enabled);
        }

        public AltObject Execute()
        {
            CommHandler.Send(cmdParams);
            var altUnityObject = CommHandler.Recvall<AltObject>(cmdParams);
            altUnityObject.CommHandler = CommHandler;
            return altUnityObject;
        }
    }
}